cobigen/cobigen-core-api/src/main/java/com/devonfw/cobigen/api/util/to/Nexus2SearchResponse.java

```diff
@@ -0,0 +1,175 @@
+
+/**
+ * Json model for nexus2 Search REST API response
+ *
+ */
+@JsonIgnoreProperties(value = { "totalCount", "from", "count", "tooManyResults", "collapsed", "repoDetails" })
+public class Nexus2SearchResponse implements SearchResponse {
+
+  /** Logger instance. */
+  @JsonIgnore
+  private static final Logger LOG = LoggerFactory.getLogger(Nexus2SearchResponse.class);
+
+  @JsonProperty("data")
+  private List<Nexus2SearchResponseData> data;
+
+  @Override
+  @JsonIgnore
+  public List<URL> getDownloadURLs() throws MalformedURLException {
+
+    List<URL> downloadLinks = new ArrayList<>();
+
+    for (Nexus2SearchResponseData item : this.data) {
+      for (Nexus2SearchResponseArtifactHits artifactHit : item.artifactHits) {
+        for (Nexus2SearchResponeArtifactLinks artifactLink : artifactHit.artifactLinks) {
+          downloadLinks.add(SearchResponseUtil.createDownloadLink(
+              MavenSearchRepositoryConstants.NEXUS2_REPOSITORY_URL + "/"
+                  + MavenSearchRepositoryConstants.NEXUS2_REPOSITORY_LINK,
+              item.getGroupId(), item.getArtifactId(), item.getVersion(), "." + artifactLink.getExtension()));
+
+        }
+      }
+    }
+
+    // removes duplicates
+    List<URL> newDownloadList = downloadLinks.stream().distinct().collect(Collectors.toList());
+
+    return newDownloadList;
+  }
+
+  @Override
+  @JsonIgnore
+  public String getJsonResponse(String repositoryUrl, String groupId) throws RESTSearchResponseException {
+
+    return getJsonResponse(repositoryUrl, groupId, null);
+  }
+
+  @Override
+  @JsonIgnore
+  public String getJsonResponse(String repositoryUrl, String groupId, String authToken)
+      throws RESTSearchResponseException {
+
+    String targetLink = repositoryUrl + "/" + MavenSearchRepositoryConstants.NEXUS2_TARGET_LINK + "?q=" + groupId;
+    LOG.info("Starting {} search REST API request with URL: {}.", getRepositoryType(), targetLink);
+
+    String jsonResponse;
+
+    jsonResponse = SearchResponseUtil.getJsonResponseStringByTargetLink(targetLink, authToken);
+
+    return jsonResponse;
+  }
+
+  @Override
+  public MavenSearchRepositoryType getRepositoryType() {
+
+    return MavenSearchRepositoryType.nexus2;
+  }
+}
+
+/**
+ *
+ * Nexus search response asset model
+ *
+ */
+@JsonIgnoreProperties(value = { "repositoryId" })
+class Nexus2SearchResponseArtifactHits {
+
+  /**
+   * artifactLinks
+   */
+  @JsonProperty("artifactLinks")
+  public List<Nexus2SearchResponeArtifactLinks> artifactLinks;
+
+}
+
+class Nexus2SearchResponeArtifactLinks {
+
+  @JsonProperty("extension")
+  private String extension;
+
+  /**
+   * @return extension
+   */
+  public String getExtension() {
+
+    return this.extension;
+  }
+
+  @JsonProperty("classifier")
+  private String classifier;
+}
+
+/**
+ *
+ * Nexus2 search response item model
+ *
+ */
+@JsonIgnoreProperties(value = { "latestRelease", "latestReleaseRepositoryId", "highlightedFragment" })
+class Nexus2SearchResponseData {
```

Comment:
same here. I think in general to be more stable in future, we should simply ignore all unknown json properties. It seems you simply exclude some of them, but this would break as soon as another property will be returned in addition

Reply:
Replaced with ignoreUnknown now.
Thanks to @mdukhan for the hint.